graphql/mutation_resolver.go

```diff
@@ -281,7 +281,11 @@ func (r *mutationResolver) EnqueuePatch(ctx context.Context, patchID string, com
 		return nil, InternalServerError.Send(ctx, fmt.Sprintf("error getting patch '%s'", patchID))
 	}
 
-	if !hasEnqueuePatchPermission(user, existingPatch) {
+	patch, err := existingPatch.ToService()
+	if err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("converting APIPatch to patch '%s'", patchID))
+	}
+	if !hasEditPatchPermission(user, patch) {
```

Comment:
It's because the project permission only checks that a user can submit patches. But we also need to check that the user is the owner of the patch so that they don't mess with other people's patches. Or they might be a superuser in which case they can do whatever they want. The function could be replaced if we added a new directive like `@requirePatchOwner`.

Reply:
Also let me know if you think there is a better name for this function!